File: retroauth-core/src/test/java/com/andretietz/retroauth/stub/TestTokenStorage.java
package com.andretietz.retroauth.stub;

import com.andretietz.retroauth.AuthenticationCanceledException;
import com.andretietz.retroauth.TokenStorage;

/**
 * Created by andre on 02.05.2016.
 */
public class TestTokenStorage implements TokenStorage<String, String, String> {

    public static final String TEST_TOKEN = "token";

    @Override
    public String createType(String[] annotationValues) {
        return null;
    }

    @Override
    public String getToken(String owner, String tokenType) throws AuthenticationCanceledException {
        return TEST_TOKEN;
    }

    @Override
    public void removeToken(String s, String s2, String s3) {

    }

    @Override
    public void storeToken(String s, String s2, String s3) {

    }
}

File: retroauth-test/src/main/java/com/andretietz/retroauth/test/TestTokenStorage.kt
package com.andretietz.retroauth.test

import com.andretietz.retroauth.Callback
import com.andretietz.retroauth.TokenStorage
import java.util.concurrent.Future

class TestTokenStorage(
  private val tokens: MutableMap<String, MutableMap<String, String>> = HashMap()
) : TokenStorage<String, String, String> {

  override fun getToken(owner: String, type: String, callback: Callback<String>?): Future<String> {
    requireNotNull(requireNotNull(tokens[owner])[type]).also {
      callback?.onResult(it)
      return TestFuture.create(it)
    }
  }

  override fun removeToken(owner: String, type: String, token: String) {
    tokens[owner]?.remove(type)
  }

  override fun storeToken(owner: String, type: String, token: String) {
    if (tokens[owner] == null) tokens[owner] = HashMap()
    if (tokens[owner]?.get(type) == null) requireNotNull(tokens[owner])[type] = token
  }
}
